synapse/replication/tcp/streams/_base.py

```diff
@@ -290,6 +290,30 @@ def __init__(self, hs):
         )
 
 
+class PresenceFederationStream(Stream):
+    """A stream used to send ad hoc presence updates over federation.
+
+    Streams the remote destination and the user ID of the presence state to
+    send.
+    """
+
+    @attr.s(slots=True, auto_attribs=True)
```

Comment:
ooooOOOOoooo pep526!

Reply:
Oh yeah, I forgot I played with this. It does have a small foot gun where if you naively do `param: Dict[..] = {}` to have a default arg it reuses the same dict (in the same way as this is a foot gun for optional params).